packages/analytics-core/src/plugins/destination.ts

```diff
@@ -179,61 +188,64 @@ export class Destination implements DestinationPlugin {
       const res = await this.config.transportProvider.send(serverUrl, payload);
       if (res === null) {
         this.fulfillRequest(list, 0, UNEXPECTED_ERROR_MESSAGE);
-        return;
+        return true;
       }
+
       if (!useRetry) {
         if ('body' in res) {
           this.fulfillRequest(list, res.statusCode, `${res.status}: ${getResponseBodyString(res)}`);
         } else {
           this.fulfillRequest(list, res.statusCode, res.status);
         }
-        return;
+        return true;
       }
-      this.handleResponse(res, list);
+
+      return this.handleResponse(res, list);
     } catch (e) {
       const errorMessage = getErrorMessage(e);
       this.config.loggerProvider.error(errorMessage);
-      this.handleResponse({ status: Status.Failed, statusCode: 0 }, list);
+      return this.handleResponse({ status: Status.Failed, statusCode: 0 }, list);
     }
   }
 
-  handleResponse(res: Response, list: Context[]) {
+  handleResponse(res: Response, list: Context[]): boolean {
     const { status } = res;
-
+    let isFullfilledRequest = false;
     switch (status) {
       case Status.Success: {
-        this.handleSuccessResponse(res, list);
+        isFullfilledRequest = this.handleSuccessResponse(res, list);
         break;
       }
       case Status.Invalid: {
-        this.handleInvalidResponse(res, list);
+        isFullfilledRequest = this.handleInvalidResponse(res, list);
         break;
       }
       case Status.PayloadTooLarge: {
-        this.handlePayloadTooLargeResponse(res, list);
+        isFullfilledRequest = this.handlePayloadTooLargeResponse(res, list);
         break;
       }
       case Status.RateLimit: {
-        this.handleRateLimitResponse(res, list);
+        isFullfilledRequest = this.handleRateLimitResponse(res, list);
         break;
       }
       default: {
         // log intermediate event status before retry
         this.config.loggerProvider.warn(`{code: 0, error: "Status '${status}' provided for ${list.length} events"}`);
-        this.handleOtherResponse(list);
         break;
       }
     }
+    return isFullfilledRequest;
   }
 
   handleSuccessResponse(res: SuccessResponse, list: Context[]) {
```

Comment:
nit: add return type. And the same for other `handleXXXresponse()` methods.
```suggestion
  handleSuccessResponse(res: SuccessResponse, list: Context[]): boolean {
```